app/forms/Transport.scala

```diff
@@ -42,15 +42,33 @@ object Transport {
 
   import ModesOfTransport._
 
-  val allowedModeOfTransport = Seq(Sea, Rail, Road, Air, PostalOrMail, FixedInstallations, InlandWaterway, Other)
+  val allowedModeOfTransport =
+    Seq(Sea, Rail, Road, Air, PostalOrMail, FixedInstallations, InlandWaterway, Other)
 
   val mapping = Forms.mapping(
     "modeOfTransport" -> requiredRadio("transport.modeOfTransport.empty")
       .verifying("transport.modeOfTransport.error", isContainedIn(allowedModeOfTransport)),
     "nationality" -> text()
       .verifying("transport.nationality.empty", nonEmpty)
-      .verifying("transport.nationality.error", isEmpty or isValidCountryCode)
+      .verifying("transport.nationality.error", isEmpty or isValidCountryCode),
+    "transportId" -> text()
+      .verifying("transport.transportId.empty", nonEmpty)
+      .verifying("transport.transportId.error", isEmpty or (noLongerThan(35) and isAlphanumeric))
   )(Transport.apply)(Transport.unapply)
 
-  def form(): Form[Transport] = Form(mapping)
+  def form: Form[Transport] =
+    Form(mapping)
+
+  def messageKey(mode: String) =
```

Comment:
In my opinion this method is terrible

Reply:
This `match` should be in separate method if we need something like this

Reply:
It should be replaced by enum, case object or anything to make it better.

Without context I'm not able to understand what are we doing here and why

Reply:
Terrible is a bit harsh.   I've had a couple of comments that the match should be extracted.  Seems excessive but I could do that.  And you are right, the problem here is to convert a 'value' to a sensible 'display text' which would have been much simpler if the values were "proper" enums.

Reply:
This method looks like method in the view, everything in the string and match inside the string. Something like this inside some specific view it can be easier to understand that you're building message keys, because something is different. In practice probably it'll be more readable to use messages key with numbers not strings, it's something for us, so `transport.modeOfTransport.1` without transformation keys. IMO if we will transform data, and make a lot of changes between values -> keys -> data or something like this it will make debugging really difficult if we'll have problem with this page